contractsSol5/KyberHintHandler.sol

```diff
@@ -336,32 +325,18 @@ contract KyberHintHandler is IKyberHint, Utils4 {
             uint failedIndex
         )
     {
-        IKyberReserve[] memory tokenToEthAddresses;
-        IKyberReserve[] memory ethToTokenAddresses;
-
         (
             tokenToEthType,
-            tokenToEthAddresses,
+            ,
+            tokenToEthReserveIds,
             tokenToEthSplits,
             ethToTokenType,
-            ethToTokenAddresses,
+            ,
+            ethToTokenReserveIds,
             ethToTokenSplits,
             failedIndex
         ) = parseHintT2T(hint);
-
-        if (failedIndex == 0) {
-            tokenToEthReserveIds = new bytes8[](tokenToEthAddresses.length);
-            ethToTokenReserveIds = new bytes8[](ethToTokenAddresses.length);
-
-            for (uint i = 0; i < tokenToEthAddresses.length; i++) {
```

Comment:
So we wanted to remove this to avoid more storage reads and additional loop, and to also remove one of the virtual functions. But this is part of an external function anyway, so we didn't care if there was additional logic.